### DISCUSSION TITLE
sslv3 alert handshake failure while running ansible-playbook

### DISCUSSION THREAD

**Discussion by Hanryfour** [aristanetworks/avd#3079] [Q&A]
Hello guys, I always get this error message  at this stage whenever  I am running ansible-playbook
Kindly assist your swift feedback would be appreciated 
I am running on unbutu 22.0,  python version 3.10.12, ansible core 2.14.0  

TASK [arista.avd.eos_config_deploy_eapi : Replace configuration with intended configuration] **********************************************
fatal: [dc1-leaf1b]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.13:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-spine1]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.10:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-leaf1a]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.12:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-spine2]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.11:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-leaf2a]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.14:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc2-spine2]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.23:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-leaf2c]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.17:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc2-spine1]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.22:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-leaf1c]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.16:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc1-leaf2b]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.15:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc2-leaf2b]: FAILED! => {"changed": false, "module_stderr": "Could not connect to https://192.168.0.25:443/command-api: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
fatal: [dc2-leaf1c]: FAILED! => {"chan

**Comment by ClausHolbechArista** [comment]
Hi, thank you for reaching out.

This is caused by SSL cipher incompatibility of Python >= 3.10 with EOS. Please check this article for details: https://arista.my.site.com/AristaCommunity/s/article/Python-3-10-and-SSLV3-ALERT-HANDSHAKE-FAILURE-error

TL;DR; add `ansible_httpapi_ciphers: "AES256-SHA:DHE-RSA-AES256-SHA:AES128-SHA:DHE-RSA-AES128-SHA"` in a group_vars file covering all your devices.
Make sure to use `ansible.netcommon >= 5.0.0`.

**Comment by ClausHolbechArista** [comment]
@Hanryfour did this solve your issue?

**Comment by Hanryfour** [comment]
Hello Claus
the issue was resolved but encountered this error  while deploying 


 
fatal: [dc1-leaf1a]: FAILED! => {"changed": false, "module_stderr": "command timeout triggered, timeout value is 30 secs.\nSee the timeout setting options in the Network Debug and Troubleshooting Guide.", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}

fatal: [dc1-leaf1b]: FAILED! => {"changed": false, "module_stderr": "command timeout triggered, timeout value is 30 secs.\nSee the timeout setting options in the Network Debug and Troubleshooting Guide.", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}

fatal: [dc1-leaf2b]: FAILED! => {"changed": false, "msg": "command timeout triggered, timeout value is 30 secs.\nSee the timeout setting options in the Network Debug and Troubleshooting Guide."}


    On Monday, 14 August 2023 at 08:21:33 BST, Claus Holbech ***@***.***> wrote:  
 
 


@Hanryfour did this solve your issue?

**Reply by ClausHolbechArista** [reply]
Hi Hanryfour
I cannot say what is wrong from this.
A few ideas:
- It could look like some kind of connectivity issue to the device?
- Maybe the configuration you pushed cut off the connection?
- Maybe a username had another password in the pushed config, so now Ansible cannot authenticate?

If you can share more information about your setup, we may be able to help.

BR
Claus
